src/api/actions.rs

```diff
@@ -114,19 +159,16 @@ impl<'octo> ActionsHandler<'octo> {
         &self,
         response: reqwest::Response,
     ) -> crate::Result<bytes::Bytes> {
-        let location = response
-            .headers()
-            .get(reqwest::header::LOCATION)
-            .expect("No Location header found in download_workflow_run_logs")
-            .to_str()
-            .expect("Location URL not valid str");
-
-        self.crab
-            ._get(location, None::<&()>)
-            .await?
-            .bytes()
-            .await
-            .context(crate::error::Http)
+        let data_response =
+            if let Some(redirect) = response.headers().get(reqwest::header::LOCATION) {
+                let location = redirect.to_str().expect("Location URL not valid str");
+
+                self.crab._get(location, None::<&()>).await?
+            } else {
+                response
+            };
+
+        data_response.bytes().await.context(crate::error::Http)
```

Comment:
Unrelated change that's a bugfix for the cases where there's no redirect header, encountered this when calling this function.